src/integration-test/java/com/commercetools/sync/integration/externalsource/shoppinglists/ShoppingListSyncIT.java

```diff
@@ -110,202 +119,173 @@ void sync_WithSameShoppingList_ShouldNotUpdateCustomer() {
     }
 
     @Test
     void sync_WithModifiedShoppingList_ShouldUpdateShoppingList() {
-        final ShoppingListDraft existingShoppingListDraft =
-                mapToShoppingListDrafts(singletonList(existingShoppingList)).get(0);
-
-        final ShoppingListDraft updatedShoppingListDraft = ShoppingListDraftBuilder.of(existingShoppingListDraft)
-                        .description(LocalizedString.ofEnglish("new-shoppinglist-description"))
-                        .slug(LocalizedString.ofEnglish("new-shoppinglist-slug"))
-                        .anonymousId("new-shoppinglist-anonymousId")
-                        .deleteDaysAfterLastModification(180)
-                        .name(LocalizedString.ofEnglish("new-shoppinglist-name"))
-                        .build();
+        final ShoppingListDraft modifiedShoppingListDraft = prepareUpdatedDraft();
 
         final ShoppingListSyncStatistics shoppingListSyncStatistics = shoppingListSync
-                .sync(singletonList(updatedShoppingListDraft))
-                .toCompletableFuture()
-                .join();
+            .sync(singletonList(modifiedShoppingListDraft))
+            .toCompletableFuture()
+            .join();
 
         assertThat(errorMessages).isEmpty();
         assertThat(warningMessages).isEmpty();
         assertThat(exceptions).isEmpty();
-        assertThat(updateActionList).contains(
-                SetDescription.of(LocalizedString.ofEnglish("new-shoppinglist-description")),
-                SetSlug.of(LocalizedString.ofEnglish("new-shoppinglist-slug")),
-                SetAnonymousId.of("new-shoppinglist-anonymousId"),
-                SetDeleteDaysAfterLastModification.of(180),
-                ChangeName.of(LocalizedString.ofEnglish("new-shoppinglist-name"))
-        );
-        assertThat(shoppingListSyncStatistics).hasValues(1, 0, 1, 0);
 
-    }
-
-    @Test
-    void sync_WithModifieCustomerReferenceInShoppingList_ShouldUpdateShoppingList() {
-
-        final ShoppingListDraft existingShoppingListDraft =
-                mapToShoppingListDrafts(singletonList(existingShoppingList)).get(0);
-
-        final Customer customer = prepareCustomer();
-        final ResourceIdentifier<Customer> customerResourceIdentifier = customer.toResourceIdentifier();
-        final ShoppingListDraft updatedShoppingListDraft = ShoppingListDraftBuilder.of(existingShoppingListDraft)
-                .customer(customerResourceIdentifier)
-                .build();
-
-        final ShoppingListSyncStatistics shoppingListSyncStatistics = shoppingListSync
-                .sync(singletonList(updatedShoppingListDraft))
-                .toCompletableFuture()
-                .join();
-
-        assertThat(errorMessages).isEmpty();
-        assertThat(warningMessages).isEmpty();
-        assertThat(exceptions).isEmpty();
-        assertThat(updateActionList).containsExactly(
-                SetCustomer.of(customer)
-        );
+        assertUpdateActions();
```

Comment:
I would actually assert that the shoppingList was updated (as your test is named). You can leave the update actions assert here, but I think the focus of the integration test shouldn't be too much on the implementation, but rather if the result is achieved.

Reply:
You're right, I am trusting here the update actions and statistics, also a bit black box testing. But could be added, let me check the effort and decide later.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/626/commits/0729ca1f3b8e70c278e640b900d0b55d930edb2c